### DISCUSSION TITLE
the agents should need to delete the docker they used

### DISCUSSION THREAD

**Discussion by byterocker20** [microsoft/autogen#1399] [Q&A]
they create too much docker, they need to delete them after using dockers
![Uploading docker.png…]()

**Comment by byterocker20** [comment]
![Uploading docker.png…]()

**Comment by gagb** [comment]
@afourney @ekzhu might have comments on this.

**Reply by afourney** [reply]
I agree. The agents leave too many images around.

AutoGenBench had the same problem, but I arranged it to auto-delete the containers and use only one base image... but it still leaves detritus if you press Ctrl-C in the middle of the run.

**Reply by ekzhu** [reply]
Thanks! We are working on the new code executor. We can add this to our roadmap issue once the current code executor PR is merged.
